app/api/routes.py

```diff
@@ -72,3 +76,29 @@ def get_languages():
         language_list = []
     finally:
         return jsonify(language_list)
+
+
+def set_resource(id, param_list, args):
+    resource = None
+    param_names = ['category', 'languages', 'name', 'url', 'paid', 'notes']
+    for index in range(len(param_names)):
+        if args.get(param_names[index]):
+            param_list[index] = args.get(param_names[index])
+    try:
+        resource = Resource.query.get(id)
+
+    except MultipleResultsFound as e:
+        print_tb(e.__traceback__)
+        print(e)
+
+    except NoResultFound as e:
+        print_tb(e.__traceback__)
+        print(e)
+
+    finally:
+        if resource:
+            for position in range(len(param_names)):
+                resource[param_names[position]] = param_list[position]
```

Comment:
After this happens, it needs to be saved in the database with `db.session.commit()`, which should go inside a try block.

Reply:
What is the error type I should specify for the `except` clause (i.e. something similar to `MultipleResultsFound` or `NoResultFound`)?

Reply:
I would just catch all SQLAlchemy errors like we do here https://github.com/OperationCode/resources_api/blob/master/app/cli.py#L52